File: MultiLineSearch/Settings/OptionPageMultilineFindReplace.vb
Imports Microsoft.VisualStudio.Shell
Imports System.Runtime.InteropServices
Imports System.ComponentModel
Imports System.Windows.Forms


Namespace Settings

    ''' <summary>
    ''' Package options
    ''' </summary>
    ''' <remarks></remarks>
    <ClassInterface(ClassInterfaceType.AutoDual), _
    Guid("E0B68D27-9489-4053-AB39-6701AC637C30")> _
    Public Class OptionPageMultilineFindReplace
        Inherits DialogPage


        Public Sub New()
            SetDefaultValues()
        End Sub


        <Browsable(False)> _
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
        Protected Overrides ReadOnly Property Window() As IWin32Window
            Get
                ' get default grid view, we will never display the page
                Return MyBase.Window
            End Get
        End Property


        Protected Overrides Sub OnApply(ByVal e As Microsoft.VisualStudio.Shell.DialogPage.PageApplyEventArgs)
            ' persist settings
            MyBase.OnApply(e)
        End Sub


        Protected Overrides Sub OnActivate(ByVal e As System.ComponentModel.CancelEventArgs)
            MyBase.OnActivate(e)
        End Sub


        ''' <summary>
        ''' Set default values.
        ''' </summary>
        ''' <remarks></remarks>
        Public Overrides Sub ResetSettings()
            SetDefaultValues()
            MyBase.ResetSettings()
        End Sub


        ''' <summary>
        ''' Set default values.
        ''' </summary>
        ''' <remarks></remarks>
        Private Sub SetDefaultValues()
            Me.IsFindOptionsCollapsed = False
            Me.IgnoreLeadingWs = False
            Me.IgnoreTrailingWs = False
            Me.IgnoreAllWs = False
            Me.SplitterPosition = 100
        End Sub


#Region "Settings properties"


        Private mIsFindOptionsCollapsed As Boolean
        ''' <summary>
        ''' Gets or sets a value indicating whether Find options section is collapsed
        ''' in Find/Search window.
        ''' </summary>
        ''' <value><see langword="true"/> if Find options section is collapsed; otherwise, <see langword="false"/>.</value>
        ''' <remarks></remarks>
        Public Property IsFindOptionsCollapsed() As Boolean
            Get
                Return mIsFindOptionsCollapsed
            End Get
            Set(ByVal value As Boolean)
                mIsFindOptionsCollapsed = value
            End Set
        End Property


        Private mIgnoreLeadingWs As Boolean
        ''' <summary>
        ''' Gets or sets a value indicating whether to ignore leading whitespaces.
        ''' </summary>
        ''' <value><see langword="true"/> if leading whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        ''' <remarks></remarks>
        Public Property IgnoreLeadingWs() As Boolean
            Get
                Return mIgnoreLeadingWs
            End Get
            Set(ByVal value As Boolean)
                mIgnoreLeadingWs = value
            End Set
        End Property


        Private mIgnoreTrailingWs As Boolean
        ''' <summary>
        ''' Gets or sets a value indicating whether to ignore trailing whitespaces.
        ''' </summary>
        ''' <value><see langword="true"/> if trailing whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        ''' <remarks></remarks>
        Public Property IgnoreTrailingWs() As Boolean
            Get
                Return mIgnoreTrailingWs
            End Get
            Set(ByVal value As Boolean)
                mIgnoreTrailingWs = value
            End Set
        End Property


        Private mIgnoreAllWs As Boolean
        ''' <summary>
        ''' Gets or sets a value indicating whether to ignore all whitespaces.
        ''' </summary>
        ''' <value><see langword="true"/> if all whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        ''' <remarks></remarks>
        Public Property IgnoreAllWs() As Boolean
            Get
                Return mIgnoreAllWs
            End Get
            Set(ByVal value As Boolean)
                mIgnoreAllWs = value
            End Set
        End Property


        Private mSplitterPosition As Integer
        ''' <summary>
        ''' Gets or sets a position (in percents) of splitter between 'Find what' and 'Replace with' text boxes.
        ''' </summary>
        ''' <value></value>
        ''' <remarks></remarks>
        Public Property SplitterPosition() As Integer
            Get
                Return mSplitterPosition
            End Get
            Set(ByVal value As Integer)
                mSplitterPosition = value
            End Set
        End Property

#End Region


    End Class

End Namespace

File: MultiLineSearch2/Settings/OptionPageMultilineFindReplace.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Microsoft.VisualStudio.Shell;
using System.Runtime.InteropServices;
using System.ComponentModel;
using System.Windows.Forms;


namespace Helixoft.MultiLineSearch.Settings
{

    /// <summary>
    /// Package options
    /// </summary>
    /// <remarks></remarks>
    [ClassInterface(ClassInterfaceType.AutoDual), Guid("E0B68D27-9489-4053-AB39-6701AC637C30")]
    public class OptionPageMultilineFindReplace : DialogPage
    {


        public OptionPageMultilineFindReplace()
        {
            SetDefaultValues();
        }


        [Browsable(false)]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        protected override IWin32Window Window
        {
            // get default grid view, we will never display the page
            get { return base.Window; }
        }


        protected override void OnApply(Microsoft.VisualStudio.Shell.DialogPage.PageApplyEventArgs e)
        {
            // persist settings
            base.OnApply(e);
        }


        protected override void OnActivate(System.ComponentModel.CancelEventArgs e)
        {
            base.OnActivate(e);
        }


        /// <summary>
        /// Set default values.
        /// </summary>
        /// <remarks></remarks>
        public override void ResetSettings()
        {
            SetDefaultValues();
            base.ResetSettings();
        }


        /// <summary>
        /// Set default values.
        /// </summary>
        /// <remarks></remarks>
        private void SetDefaultValues()
        {
            this.IsFindOptionsCollapsed = false;
            this.IgnoreLeadingWs = false;
            this.IgnoreTrailingWs = false;
            this.IgnoreAllWs = false;
            this.SplitterPosition = 100;
        }


        #region "Settings properties"


        private bool mIsFindOptionsCollapsed;
        /// <summary>
        /// Gets or sets a value indicating whether Find options section is collapsed
        /// in Find/Search window.
        /// </summary>
        /// <value><see langword="true"/> if Find options section is collapsed; otherwise, <see langword="false"/>.</value>
        /// <remarks></remarks>
        public bool IsFindOptionsCollapsed
        {
            get { return mIsFindOptionsCollapsed; }
            set { mIsFindOptionsCollapsed = value; }
        }


        private bool mIgnoreLeadingWs;
        /// <summary>
        /// Gets or sets a value indicating whether to ignore leading whitespaces.
        /// </summary>
        /// <value><see langword="true"/> if leading whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        /// <remarks></remarks>
        public bool IgnoreLeadingWs
        {
            get { return mIgnoreLeadingWs; }
            set { mIgnoreLeadingWs = value; }
        }


        private bool mIgnoreTrailingWs;
        /// <summary>
        /// Gets or sets a value indicating whether to ignore trailing whitespaces.
        /// </summary>
        /// <value><see langword="true"/> if trailing whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        /// <remarks></remarks>
        public bool IgnoreTrailingWs
        {
            get { return mIgnoreTrailingWs; }
            set { mIgnoreTrailingWs = value; }
        }


        private bool mIgnoreAllWs;
        /// <summary>
        /// Gets or sets a value indicating whether to ignore all whitespaces.
        /// </summary>
        /// <value><see langword="true"/> if all whitespaces are ignored; otherwise, <see langword="false"/>.</value>
        /// <remarks></remarks>
        public bool IgnoreAllWs
        {
            get { return mIgnoreAllWs; }
            set { mIgnoreAllWs = value; }
        }


        private int mSplitterPosition;
        /// <summary>
        /// Gets or sets a position (in percents) of splitter between 'Find what' and 'Replace with' text boxes.
        /// </summary>
        /// <value></value>
        /// <remarks></remarks>
        public int SplitterPosition
        {
            get { return mSplitterPosition; }
            set { mSplitterPosition = value; }
        }

        #endregion


    }

}
